common/p_map.cpp

```diff
@@ -697,12 +697,31 @@ static BOOL PIT_CheckThing (AActor *thing)
 		// [SL] Work-around the additional height added to players
 		// in P_CheckPosition. Don't let players grab items above
 		// their real height!
-		if (!P_AllowPassover() || !tmthing->player ||
-			thing->z < tmthing->z + tmthing->height - 24*FRACUNIT)
+		if (!P_AllowPassover() || thing->z < tmthing->z + tmthing->height - 24*FRACUNIT)
 			P_TouchSpecialThing (thing, tmthing);	// can remove thing
+
+		return !solid;
 	}
 
-	return !solid;
+	// killough 3/16/98: Allow non-solid moving objects to move through solid
+	// ones, by allowing the moving thing (tmthing) to move if it's non-solid,
+	// despite another solid thing being in the way.
+	// killough 4/11/98: Treat no-clipping things as not blocking
+	// ...but not in demo_compatibility mode
+
+	// e6y
+	// Correction of wrong return value with demo_compatibility.
+	// There is no more synch on http://www.doomworld.com/sda/dwdemo/w303-115.zip
+	// (with correction in setMobjInfoValue)
+	if (demoplayback || !co_boomphys
+	    //&& !prboom_comp[PC_TREAT_NO_CLIPPING_THINGS_AS_NOT_BLOCKING].state
```

Comment:
minor: it's preferable to not add commented out code. If you think it'd be helpful to add some breadcrumbs for finding this original Boom code in case we add Boom compat levels in the future, that'd would be helpful

Reply:
That was the intent, eventually I am going to go back and re-engineer to allow boom comp levels one day.